node/bft/src/sync/mod.rs

```diff
@@ -489,11 +489,12 @@ impl<N: Network> Sync<N> {
             let commit_round = leader_certificate.round();
             let certificate_round = commit_round.saturating_add(1);
```

Comment:
I am curious why we use `saturating_add` in places like this, and if a `checked_add` makes more sense. If the round number overflows, something is really wrong, but this code will just silently continue.

Reply:
Yes, a plain `+` would be simpler, faster, and more readable, here and in many other places. The whole thing would fail catastrophically if we ever ran out of round numbers, or block heights, so there is an underlying assumption that we'll always be well below the max's, for the lifetime of the implementation.

Reply:
I guess a `checked_add` could protect against potential implementation errors in release mode, although I don't think that the invariants about always being well below the max should be difficult to establish (informally).

Reply:
I've pushed a commit that changes it to a checked_add.